memory.c

```diff
@@ -263,6 +264,83 @@ allocator_print(Allocator * alloc)
     }
 }
 
+void *
+allocator_realloc_int(Allocator * alloc, void * ptr, size_t new_size, char * fmt, ...)
```

Comment:
resize instead of realloc, since the ptr is guaranteed not to move.

Reply:
Only for bottom allocations.

Reply:
I see...